mesonbuild/interpreter/interpreter.py

```diff
@@ -2467,33 +2467,35 @@ def func_install_data(self, node: mparser.BaseNode,
                     '"rename" and "sources" argument lists must be the same length if "rename" is given. '
                     f'Rename has {len(rename)} elements and sources has {len(sources)}.')
 
+        install_dir = kwargs['install_dir']
+        if not install_dir:
+            subdir = self.active_projectname
+            install_dir = P_OBJ.OptionString(os.path.join(self.environment.get_datadir(), subdir), os.path.join('{datadir}', subdir))
+
         install_mode = self._warn_kwarg_install_mode_sticky(kwargs['install_mode'])
-        return self.install_data_impl(sources, kwargs['install_dir'], install_mode,
-                                      rename, kwargs['install_tag'],
+        return self.install_data_impl(sources, install_dir,
+                                      install_mode, rename,
+                                      kwargs['install_tag'],
```

Comment:
I don't think there's a particular reason to split this up across lines?

Reply:
It is to align it with how the arguments are split on multiple lines in the function definition. I found it easier this way. But I can rewrap it.

Reply:
I don't think they need to match.

The original reason I mentioned it was because the only *effective* change is:
```diff
-        return self.install_data_impl(sources, kwargs['install_dir'], install_mode,
+        return self.install_data_impl(sources, install_dir, install_mode,
```

and in general I think very short lines are a mistake. So I'd either want to keep the lines the same but delete some characters from the first line only, or else reformat it in the process to be more compact. It feels strange to make it more spread out instead. ;)

The function signature is laboring under the requirement to have type annotations while still avoiding egregious line length overflow, and indeed if it were all on one line it would be 300 characters. :D

Probably here I'd suggest this diff:
```diff
         install_mode = self._warn_kwarg_install_mode_sticky(kwargs['install_mode'])
-        return self.install_data_impl(sources, kwargs['install_dir'], install_mode,
-                                      rename, kwargs['install_tag'],
+        return self.install_data_impl(sources, install_dir, install_mode, rename, kwargs['install_tag'],
                                       preserve_path=kwargs['preserve_path'])
```

Reply:
Done